### PR TITLE
systemd-run: add --slice-inherit

### PR DESCRIPTION
**Pull request by elmarco**
Add a new option to easily place a slice within the systemd-run slice.

Obsoletes: #15244

### CODE REVIEW

File: src/run/run.c [hunk 1]
```diff
@@ -194,6 +198,7 @@ static int parse_argv(int argc, char *argv[]) {
                 { "unit",              required_argument, NULL, 'u'                   },
                 { "description",       required_argument, NULL, ARG_DESCRIPTION       },
                 { "slice",             required_argument, NULL, ARG_SLICE             },
+                { "slice-inherit",     optional_argument, NULL, ARG_SLICE_INHERIT     },
```
Comment (poettering) [inline src/run/run.c]: hmm, getopt()'s `optional_argument` options are not nice to use, since they have this weird logic where they might or might not eat up the next argument. we are nowadays pretty careful with introducing new options that use this. e.g.:

```
systemd-run --slice-inherit foobar baz
```

if you look at the line above it could mean that you call the command the switch with no argument, and foobar and baz as separate comand line arguments. Or it could be the switch with foobar as argument, and only baz as separate command line argument. The glibc getopt code implements the second one, but it's not obvious to the reader I think.

Long story short. what about making `--slice-inherit` the long option unconditionally with argument, and then introuduce a short version `-I` or so that unconditionally takes no argument. so that you choose the version with/without argument by using the long/short option. 

Does that make sense?

File: man/systemd-run.xml [hunk 2]
```diff
@@ -151,6 +151,19 @@
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--slice-inherit</option></term>
+        <term><option>--slice-inherit=</option></term>
+
+        <listitem><para>Make the new <filename>.service</filename> or <filename>.scope</filename> unit part
+        of the specified inherited slice, instead of <filename>system.slice</filename>.</para>
```
Comment (poettering) [inline man/systemd-run.xml]: in user mode the services are put in the root slice by default

File: src/run/run.c [hunk 3]
```diff
@@ -637,23 +647,58 @@ static int transient_unit_set_properties(sd_bus_message *m, UnitType t, char **p
...
 
 static int transient_cgroup_set_properties(sd_bus_message *m) {
+        _cleanup_free_ char *slice = NULL;
+        _cleanup_free_ char *name = NULL;
         int r;
         assert(m);
 
-        if (!isempty(arg_slice)) {
-                _cleanup_free_ char *slice = NULL;
+        if ((int) !!arg_slice + (int) !!arg_slice_inherit_set > 1)
```
Comment (poettering) [inline src/run/run.c]: why not `if (arg-slice && arg_slice_inherit_set)`? Would be more readable to me...

File: src/run/run.c [hunk 4]
```diff
@@ -637,23 +647,58 @@ static int transient_unit_set_properties(sd_bus_message *m, UnitType t, char **p
...
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "--slice may not be combined with --slice-inherit.");
```
Comment (poettering) [inline src/run/run.c]: we generally try to suffix options that take options with `=` in docs and log msgs, hence `--slice` → `--slice=` here please

File: src/run/run.c (hunk 1 above)
Comment (poettering) [reply src/run/run.c]: or, another idea: make --slice-inherit a switch without any arguments, and allow it to be combined with --slice=. If they are combined, then --slice= is the suffix appended to the caller's slice.

I think this would make a ton of sense, no?

File: man/systemd-run.xml (hunk 2 above)
Comment (elmarco) [reply man/systemd-run.xml]: I copied the doc from --slice. Could you perhaps fix the --slice doc for --user mode? Cause I am not sure what's happening or supposed to happen.

File: src/run/run.c (hunk 1 above)
Comment (elmarco) [reply src/run/run.c]: I have not a strong systemd-culture :) so whatever you think fits best.

File: src/run/run.c (hunk 3 above)
Comment (elmarco) [reply src/run/run.c]: copied from other argument handling style, but sure.

File: src/run/run.c (hunk 4 above)
Comment (elmarco) [reply src/run/run.c]: oh I didn't notice, again I looked at other errors above, but they didn't take argument I guess.

**Comment by poettering** [comment]
lgtm, thanks

will prep doc fix for --slice= as requested

**Comment by poettering** [comment]
see  #15367 for the promised fix
